jwql/instrument_monitors/nirspec_monitors/ta_monitors/wata_monitor.py

```diff
@@ -0,0 +1,600 @@
+                          'EXTCOLST': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'stamp_start_col'},
+                          'EXTROWST': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'stamp_start_row'},
+                          'TA_DTCTR': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'star_detector'},
+                          'BOXPKVAL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'max_val_box'},
+                          'BOXPKCOL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'max_val_box_col'},
+                          'BOXPKROW': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'max_val_box_row'},
+                          'TA_ITERS': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'iterations'},
+                          'CORR_COL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'corr_col'},
+                          'CORR_ROW': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'corr_row'},
+                          'IMCENCOL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'stamp_final_col'},
+                          'IMCENROW': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'stamp_final_row'},
+                          'DTCENCOL': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'detector_final_col'},
+                          'DTCENROW': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'detector_final_row'},
+                          'SCIXCNTR': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'final_sci_x'},
+                          'SCIYCNTR': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'final_sci_y'},
+                          'TARGETV2': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'measured_v2'},
+                          'TARGETV3': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'measured_v3'},
+                          'V2_REF': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'ref_v2'},
+                          'V3_REF': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'ref_v3'},
+                          'V2_RESID': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'v2_offset'},
+                          'V3_RESID': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'v3_offset'},
+                          'SAM_X': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'sam_x'},
+                          'SAM_Y': {'loc': 'ta_hdr', 'alt_key': None, 'name': 'sam_y'}
+                          }
+        # fill out the dictionary  to create the dataframe
+        wata_dict = {}
+        for fits_file in new_filenames:
+            wata_info = self.get_tainfo_from_fits(fits_file)
+            if wata_info is None:
+                continue
+            main_hdr, ta_hdr = wata_info
+            for key, key_dict in keywds2extract.items():
+                key_name = key_dict['name']
+                if key_name not in wata_dict:
+                    wata_dict[key_name] = []
+                ext = main_hdr
+                if key_dict['loc'] == 'ta_hdr':
+                    ext = ta_hdr
+                try:
+                    val = ext[key]
+                except:
```

Comment:
@penaguerrero do you know what exception you are trying to capture here? Is it a `KeyError`?